addons/account/models/account_move_line.py

```diff
@@ -2094,6 +2094,7 @@ def _prepare_exchange_difference_move_vals(self, amounts_list, company=None, exc
                     'name': _('Currency exchange rate difference'),
                     'debit': amount_residual if amount_residual > 0.0 else 0.0,
                     'credit': -amount_residual if amount_residual < 0.0 else 0.0,
+                    'analytic_distribution': kwargs.get('exchange_analytic_distribution', False),
```

Comment:
as discussed: `False` might prevent computations, so let's only add the key/value if the distribution is given